server/grpc/grpc_web.go

```diff
@@ -28,8 +30,18 @@ func StartGRPCWeb(grpcSrv *grpc.Server, config config.Config) (*http.Server, err
 		Addr:    config.GRPCWeb.Address,
 		Handler: http.HandlerFunc(handler),
 	}
-	if err := grpcWebSrv.ListenAndServe(); err != nil {
+
+	errCh := make(chan error)
+	go func() {
+		if err := grpcWebSrv.ListenAndServe(); err != nil {
+			errCh <- fmt.Errorf("failed to serve: %w", err)
+		}
+	}()
+
+	select {
+	case err := <-errCh:
 		return nil, err
+	case <-time.After(5 * time.Second): // assume server started successfully
```

Comment:
Isn't 5s too long?

Reply:
3 might be ok

Reply:
5 seconds is also used for several other servers, do you think we should change them all, maybe share a same constant variable.

Reply:
done

Reply:
5s seams a lot, I think we can do faster, but we should check with some ops guy. Maybe @anilCSE  can provide some input?